src/components/application_manager/src/application_impl.cc

```diff
@@ -399,6 +399,33 @@ bool ApplicationImpl::audio_streaming_allowed() const {
   return audio_streaming_allowed_;
 }
 
+bool ApplicationImpl::SetVideoConfig(protocol_handler::ServiceType service_type,
+                                     const smart_objects::SmartObject& params) {
+  using namespace protocol_handler;
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (ServiceType::kMobileNav == service_type) {
+    // See StartStreaming(). We issue SetVideoConfig and StartStream
+    // only when streaming is not approved yet
+    if (!video_streaming_approved()) {
```

Comment:
Yes, that's correct. That is why the if statement says "if the video is NOT approved" then we send a SetVideoConfig request. Let me know if I missed something.